packages/compute-baseline/src/browser-compat-data/feature.test.ts

```diff
@@ -69,5 +69,86 @@ describe("features", function () {
+        const cr = browser("chrome");
+        assert.equal(
+          feature("api.Attr").flatSupportedIn(cr.version("100")),
+          true,
+        );
+      });
+
+      it("returns false for qualified support", function () {
+        const cr = browser("chrome");
+        assert.equal(
+          feature("css.properties.line-clamp").flatSupportedIn(
+            cr.version("100"),
+          ),
+          false,
+        ); // { version_added: "…", "prefix": "-webkit-" }
+      });
+
+      it("returns false for wholly unsupported", function () {
+        const fx = browser("firefox");
+        assert.equal(
+          feature("api.Accelerometer").flatSupportedIn(fx.current()),
+          false,
+        ); // { version_added: false }
+      });
+
+      it("returns null for unknown support", function () {
+        const edge = browser("edge");
+        const f = feature("svg.elements.animate"); // { version_added: "≤79" }
+
+        assert.equal(f.flatSupportedIn(edge.version("12")), null);
+        assert.equal(f.flatSupportedIn(edge.version("79")), true);
+        assert.equal(f.flatSupportedIn(edge.version("80")), true);
+      });
+    });
+
+    describe("supportedIn()", function () {
+      it("returns support for features supported with and without qualification", function () {
+        const compat = new Compat();
+        const cr = browser("chrome");
+
+        // { version_added: "…" }
+        const bgColor = feature("css.properties.background-color").supportedIn(
+          cr.version("100"),
+        );
+        assert.equal(bgColor.length, 1);
+        assert.equal(bgColor[0]?.supported, true);
+
+        // { version_added: "…", prefix: "-webkit-" }
+        const lineClamp = feature("css.properties.line-clamp").supportedIn(
+          cr.version("100"),
+        );
+        assert.equal(lineClamp.length, 1);
+        assert.equal(lineClamp[0]?.supported, true);
+        assert.equal(lineClamp[0]?.qualifications?.prefix, "-webkit-");
+      });
+
+      it("returns mixed results for (un)prefixedfeatures", function () {
+        const fx = browser("firefox");
+        const actual = feature(
+          "css.types.image.gradient.repeating-linear-gradient",
+        ).supportedIn(fx.version("100"));
+        assert.equal(actual.length, 3); // unprefixed, -moz-, and -webkit-
+        assert(actual.some((s) => s.supported && "qualifications" in s));
+        assert(actual.some((s) => s.supported && !("qualifications" in s)));
+      });
+
+      it("returns unknown support before version ranges", function () {
+        const edge = browser("edge");
+        const f = feature("svg.elements.animate");
+        const unknown = f.supportedIn(edge.version("12"));
+        assert.equal(unknown.length, 1);
+        assert.equal(unknown[0]?.supported, null);
+
+        const known = f.supportedIn(edge.version("79"));
+        assert.equal(known.length, 1);
+        assert.equal(known[0]?.supported, true);
+      });
+    });
```

Comment:
To find good test cases for the flat version, I ended up needing an un-flat version. We can dump this if we like, if we think we won't use it. OTOH, I did use it to write test cases. A conundrum.

Reply:
Unless it gets exported to compute-baseline users, let's have it. Although if the flat version is the one we'll use most, maybe give that the nice name and rename the other one to a longer name?

Reply:
Done with c07b62db0f812f87bb3e51d8f7091692239fb12d.